Remotion/Data/DomainObjects/ReflectionUtility.cs

```diff
@@ -262,11 +266,16 @@ public static Type GetObjectListTypeParameter (Type type)
     {
       ArgumentUtility.CheckNotNull ("type", type);
 
-      var typeParameters = type.GetAscribedGenericArguments (typeof (ObjectList<>));
-      var typeParameter = typeParameters[0];
-      if (typeParameter.IsGenericParameter)
-        return null;
-      return typeParameter;
+      Type ValueFactory (Tuple<Type, Type> key)
+      {
+        var (type, objectListType) = key;
+        var typeParameter = type.GetAscribedGenericArguments (objectListType)[0];
+        return typeParameter.IsGenericParameter ? null : typeParameter;
+      }
+
+      s_ascribedGenericArgumentsCacheValueFactory ??= ValueFactory;
```

Comment:
We should check what the compiler does. There's rules regarding what's getting automatic delegate caching code generated by the c# compiler.